check_copyright.py

```diff
@@ -501,10 +509,22 @@ def check_copyrights(args: argparse.Namespace, config: configparser.ConfigParser
             print(f'{TERMINAL_GRAY}"{file_name}" is using config section "{matched_section}" which does not perform the check! Skipping.{TERMINAL_RESET}')
             continue
 
+        try:
+            # Run a git log command to find out if the file being processed has been newly added to the git index
+            git_log_output = subprocess.check_output(['git', 'log', '--follow', '--', file_name], universal_newlines=True)
+
+            # If the git commit history is empty, it is a new file
+            if git_log_output:
+                is_new_file = False
```

Comment:
```python
is_new_file = not git_log_output
```

Reply:
Added the following change instead to the same effect-
```
                if is_new_file:
                    years = (0, None)
                else:
                    years = extract_years_from_espressif_notice(matches.group(1))
```